src/libcrun/linux.c

```diff
@@ -3101,6 +3111,52 @@ init_container (libcrun_container_t *container, int sync_socket_container, struc
   int ret;
   const char success = 0;
 
+  if (init_status->idx_pidns_to_join_immediately >= 0 || init_status->idx_timens_to_join_immediately >= 0)
+    {
+      pid_t new_pid;
+
+      if (init_status->idx_pidns_to_join_immediately >= 0)
+        {
+          ret = setns (init_status->fd[init_status->idx_pidns_to_join_immediately], CLONE_NEWPID);
```

Comment:
is there any risk of this being out of bounds?

Reply:
it is initialized to -1 and the only place where we set it is in the `for (i = 0; i < init_status.fd_len; i++)` loop so it should not be possible to set it to a value higher than fd_len.  If you'd like we can add an extra check to make sure